tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php

```diff
@@ -518,23 +518,14 @@ function foo(A $a, \Other\B $b) {}
         yield '@link shall not crash fixer' => [
             '<?php
 
-use Symfony\Component\Validator\Constraints\Valid;
-/**
- * {@link \Symfony\Component\Validator\Constraints\Valid} is assumed.
- *
- * @return void
- */
-function validate(): void {}
-',
-            '<?php
-
 /**
  * {@link \Symfony\Component\Validator\Constraints\Valid} is assumed.
  *
  * @return void
  */
 function validate(): void {}
 ',
+            null,
             ['import_symbols' => true],
```

Comment:
pls do not cut down this test.

Reply:
The test is not cut, the fixer was actually fixed - `use` was wrongly added but not actually used

Reply:
it was in the `@link`.

we count-in the usage in phpdocs [eg https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/b87fabcf1dd1993c8b1319f218deea660e050cce/tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php#L1313 ]

Reply:
please see the https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7649/files - I only see 2 added tests and 1 test where `use` was added wrongly (as later not used/replaced) - this is now fixed and the "expected test string" was removed and now the string is the same as "input test string".

Reply:
I do understand the change you done to this test.
On master, we are doing the import of Valid::class, here we do not.

If it would be `@param \Foo\Valid`, we would import it and convert annotation to `@param Valid`.
Yet it is `@link \Foo\Valid`, where we (on master) properly import it, but not-yet convert to `@link Valid`. I would rather add missing `@link Valid` conversion than stop importing, as your PR doing now.

Reply:
reason why I still think the change is legit:
- `@link` is not even part of the `phpdoc_tags` option
- `* {@xxx ` is not even valid phpdoc (notice `{`)
- previously, the FQCN was imported but never used, this reason itself should be enough :)

Reply:
our own Fixer considers usage valid and recognised `Valid` class inside `{@link...}` as usage, thus do not remove the `use` import.

```
ker@d ~/github/PHP-CS-Fixer λ ./php-cs-fixer check x.php -vvv --diff --rules=no_unused_imports
PHP CS Fixer 3.45.1-DEV Three Keys by Fabien Potencier and Dariusz Ruminski.
PHP runtime: 8.3.1
Loaded config default from "/Users/d.ruminski/github/PHP-CS-Fixer/.php-cs-fixer.dist.php".
Paths from configuration file have been overridden by paths provided as command arguments.
 1/1 [████████████████████████████] 100%

   1) x.php (no_unused_imports)
      ---------- begin diff ----------
--- /Users/d.ruminski/github/PHP-CS-Fixer/x.php
+++ /Users/d.ruminski/github/PHP-CS-Fixer/x.php
@@ -1,7 +1,6 @@
 <?php
 
 use Symfony\Component\Validator\Constraints\Valid;
-use Foo\Bar;
 
 /**
  * {@link Valid} is assumed.
  */
 function validate(): void {}

      ----------- end diff -----------
```

I understand that you would like this behaviour to be different, but please keep the original variant of this test case.

Reply:
I would like to help, but your requirement is of the scope of this PR.

There are at 3 different isolated reasons I mentioned in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7649#discussion_r1439097812. We cannot import any FQCN from a free text (whole phpdoc). Originally, it was working by a coincidence. If you have any suggestion, feel free to push into my branch directly.

Reply:
I'm out of capacity to handle it right now.
But that also doesn't mean we should merge with braking existing behaviour.

Reply:
Are we really on the same page? The test is described to test if the fixer does not crash. Also, the only change is the symbol is no longer imported - previously it was, but was not used, what is the purpose of importing unused symbol?

Reply:
`phpdoc_tags` option defines parseable tags, `@link` (or more precisely `link`) tag is NOT present, why such tag should be touch? Why do you insist on importing it even if the tag was/is not used (the import is/was useless)?

Reply:
I see nothing more I can add than already shared via https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7649#discussion_r1439099726

Reply:
In that code you showed `no_unused_imports`. Such fixer continues to work perfectly and it works defensively - it is for code removal.

This fixer adds code and it should work on correctly parsed & allowed phpdoc tags. This is what this PR is about. Therefore, I am asking you to describe the behaviour and help me to satisfy both sides. Please understand, ATM, I do not know exactly why you defending this fix and how I can address your feedback.

Reply:
fixed - I have implemented inline phpdoc support for this fixed based on https://docs.phpdoc.org/2.9/references/phpdoc/tags/link.html grammar (`{\h*@tag ...}`)

Reply:
Thanks! yep, most of the fixers are either based on phpdocs from phpdocumentator or forever-draft psr5